rubix-common/src/main/java/com/qubole/rubix/common/metrics/CustomMetricsReporterProvider.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.common.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+import com.qubole.rubix.spi.CacheConfig;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.qubole.rubix.common.utils.ClusterUtil.getMetricsReportors;
+
+public class CustomMetricsReporterProvider {
```

Comment:
shouldn't this be a singleton instead? and not a member variable in it.

Reply:
done